business/istio_status.go

```diff
@@ -0,0 +1,194 @@
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+)
+
+// SvcService deals with fetching istio/kubernetes services related content and convert to kiali model
+type IstioStatusService struct {
+	k8s kubernetes.IstioClientInterface
+}
+
+type ComponentStatus struct {
+	// The app label value of the Istio component
+	//
+	// example: istio-ingressgateway
+	// required: true
+	Name string `json:"name"`
+
+	// The status of a Istio component
+	//
+	// example:  Not Found
+	// required: true
+	Status string `json:"status"`
+
+	// When true, the component is necessary for Istio to function. Otherwise, it is an addon
+	//
+	// example:  true
+	// required: true
+	IsCore bool `json:"is_core"`
+}
+
+type IstioComponentStatus struct {
+	// Message regarding the Istio architecture: monolith, mixer-based, multiple or none
+	//
+	// example: Istio status disabled: multiple pilots found
+	Message string `json:"message,omitempty"`
+
+	// List of each istio deployment status
+	//
+	// required: true
+	List []ComponentStatus `json:"list"`
+}
+
+const (
+	Healthy   string = "Healthy"
+	Unhealthy string = "Unhealthy"
+	NotFound  string = "NotFound"
+)
+
+// List of workloads part of a Istio deployment and if whether it is mandatory or not.
+// It follows the default profile
+var components = map[string]map[string]bool{
+	"monolith": {
+		"istio-egressgateway":  false,
+		"istio-ingressgateway": true,
+		"istiod":               true,
+		"grafana":              false,
+		"istio-tracing":        false,
+		"prometheus":           true,
+	},
+	"mixer": {
+		"istio-citadel":          true,
+		"istio-egressgateway":    false,
+		"istio-galley":           true,
+		"istio-ingressgateway":   true,
+		"istio-pilot":            true,
+		"istio-policy":           true,
+		"istio-sidecar-injector": true,
+		"istio-telemetry":        true,
+		"grafana":                false,
+		"istio-tracing":          false,
+		"prometheus":             true,
+	},
+}
+
+func (iss *IstioStatusService) GetStatus() (IstioComponentStatus, error) {
+	if !config.Get().ExternalServices.Istio.IstioStatusEnabled {
+		return IstioComponentStatus{}, nil
+	}
+
+	// Fetching workloads from control plane namespace
+	ds, error := iss.k8s.GetDeployments(config.Get().IstioNamespace)
+	if error != nil {
+		return IstioComponentStatus{}, error
+	}
+
+	arch := iss.detectArchitecture(ds)
+	if unex, isc := unexpectedArch(arch); unex {
+		return isc, nil
+	}
+
+	return iss.getStatusOf(arch, ds)
+}
+
+func unexpectedArch(arch string) (bool, IstioComponentStatus) {
+	unex, isc := false, IstioComponentStatus{}
+
+	if arch == "multiple" {
+		unex, isc = true, IstioComponentStatus{
+			Message: "Istio Status disabled: Multiple Pilot found",
+		}
+	} else if arch == "notfound" {
+		unex, isc = true, IstioComponentStatus{
+			Message: "Istio Status disabled: Pilot not found",
+		}
+	}
+
+	return unex, isc
+}
+
+func (iss *IstioStatusService) detectArchitecture(ds []apps_v1.Deployment) string {
```

Comment:
Instead of this, is it possible to use what is being done for telemetry v1: https://github.com/kiali/kiali/pull/2682 ?
Or is it different use cases that I don't see?

(I'm suggesting that they rename func `IstioSupportsCanonical` to `IsMixerEnabled`, so that's it's more appropriate here)

Reply:
Does enabling telemetry v1 on Istio 1.5 means that also non telemetry related components (citadel, galley, injector, etc) are deployed? Which are the deployments added into the 1.5 installation to support telemetry v1?

I think that the new method from #2682 might help to add telemetry deployments to `istiod`-like installation. But I'd need to install 1.5 with that flag changed to see new deployments added (if they are).